pyriemann/spatialfilters.py

```diff
@@ -467,3 +469,266 @@ def fit(self, X, y):
+
+    It can be apply:
+        - on a single subject, to solve the classical BSS problem [1],
+        - on several subjects, to solve the group BSS (gBSS) problem [2],
+        - on several experimental conditions (for eg, baseline versus task).
+
+    AJDC estimates Fourier cospectral matrices by the Welch's method, and
+    applies a trace-normalization. If necessary, it averages cospectra across
+    subjects, and concatenates them along experimental conditions.
+    Then, a dimension reduction and a whitening are applied on cospectra.
+    An approximate joint diagonalization (AJD) [3] allows to estimate the joint
+    diagonalizer, not constrained to be orthogonal. Finally, forward and
+    backward spatial filters are computed.
+
+    Parameters
+    ----------
+    window : int (default 128)
+        The length of the FFT window used for spectral estimation.
+    overlap : float (default 0.5)
+        The percentage of overlap between window.
+    fmin : float | None , (default None)
+        The minimal frequency to be returned. Since BSS models assume zero-mean
+        processes, the first cospectrum (0 Hz) must be excluded.
+    fmax : float | None , (default None)
+        The maximal frequency to be returned.
+    fs : float | None , (default None)
+        The sampling frequency of the signal.
+    expl_var : float (default 0.999)
+        The percentage of explained variance in ]0, 1], for dimension reduction
+        of cospectra, because Pham's AJD is sensitive to matrices conditioning.
+    verbose : bool (default True)
+        Verbose flag.
+
+    Attributes
+    ----------
+    n_channels_ : int
+        If fit, the number of channels of the signal.
+    freqs_ : ndarray, shape (n_freqs,)
+        If fit, the frequencies associated to cospectra.
+    n_sources_ : int
+        If fit, the number of components of the source space.
+    forward_filters_ : ndarray, shape ``(n_sources_, n_channels_)``
+        If fit, the spatial filters used to transform signal into source,
+        also called deximing or separating matrix.
+    backward_filters_ : ndarray, shape ``(n_channels_, n_sources_)``
+        If fit, the spatial filters used to transform source into signal,
+        also called mixing matrix.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.7.dev
+
+    See Also
+    --------
+    CospCovariances
+
+    References
+    ----------
+    [1] M. Congedo, C. Gouy-Pailler, C. Jutten, "On the blind source separation
+    of human electroencephalogram by approximate joint diagonalization of
+    second order statistics", Clin Neurophysiol, 2008
+
+    [2] M. Congedo, R. John, D. De Ridder, L. Prichep, "Group indepedent
+    component analysis of resting state EEG in large normative samples",
+    Int J Psychophysiol, 2010
+
+    [3] D.-T. Pham, "Joint approximate diagonalization of positive definite
+    Hermitian matrices", SIAM J Matrix Anal Appl, 2001
```

Comment:
can you use references with url links as requested in https://github.com/pyRiemann/pyRiemann/pull/113/files ?

Reply:
Ok, but I don't know why it creates a line break in the doc.